implementation/src/main/java/io/smallrye/config/SmallRyeConfig.java

```diff
@@ -224,13 +211,37 @@
 
         String value = configValue.getValue(); // Can return the empty String (which is not considered as null)
 
+        return convertValue(name, value, converter);
+    }
+
+    /**
+     * 
+     * This method handles converting values for both CDI injections and programatical calls.<br>
+     * <br>
+     * 
+     * Calls for converting non-optional values ({@link Config#getValue} and "Injecting Native Values")
+     * should throw an {@link Exception} for each of the following:<br>
+     * 
+     * 1. {@link IllegalArgumentException} - if the property cannot be converted by the {@link Converter} to the specified type
+     * <br>
+     * 2. {@link NoSuchElementException} - if the property is not defined <br>
+     * 3. {@link NoSuchElementException} - if the property is defined as an empty string <br>
+     * 4. {@link NoSuchElementException} - if the {@link Converter} returns {@code null} <br>
+     * <br>
+     * 
+     * Calls for converting optional values ({@link Config#getOptionalValue} and "Injecting Optional Values")
+     * should only throw an {@link Exception} for #1 ({@link IllegalArgumentException} when the property cannot be converted to
+     * the specified type).
+     */
+    public static <T> T convertValue(String name, String value, Converter<T> converter) {
```

Comment:
Arguably, this shouldn't be a public method at all. If it wasn't called from `ConfigProducerUtil` it could be private. Would it be better to have this method in a separate class anyway?

If you do want it in the API, then I'm in two minds. At the moment it doesn't use any state at all and it can be quite nice to know that a method operates only on its parameters. It confirms that it really is that simple and it makes it easier to call. On the other hand, if it's behaviour could change and it might need the object state in the future, then it might be better to make it non-static now.

Reply:
> Arguably, this shouldn't be a public method at all. If it wasn't called from `ConfigProducerUtil` it could be private. Would it be better to have this method in a separate class anyway?
> 

We could move it to another class, but we don't have a suitable one. I think it doesn't fit with `Converters` and I'm trying to keep the class count low, since we are mostly single package :)

> If you do want it in the API, then I'm in two minds. At the moment it doesn't use any state at all and it can be quite nice to know that a method operates only on its parameters. It confirms that it really is that simple and it makes it easier to call. On the other hand, if it's behaviour could change and it might need the object state in the future, then it might be better to make it non-static now.

I agree with your statement, but in this case even if the method only operates in its parameter, we should make it not static, for consistency with the rest of the API, plus we don't really have any other static methods there.

Reply:
I have made the method non-static